### PR TITLE
Implement BoundaryVisibility

### PR DESCRIPTION
**Pull request by FejZa**
# XRTK - Mixed Reality Toolkit Pull Request

## Overview

Added a new enum type `BoundaryVisibility`. Mostly to be able to specify the `BoundaryVisibility.Unknown` state, which is useful when we don't know what the current visibility state is.

## Breaking Changes

- New IMixedRealityBoundaryDataProvider.BoundaryVisibility property instead of bool type

## Related Submodule Changes

https://github.com/XRTK/WindowsMixedReality/pull/132

### CODE REVIEW

File: XRTK-Core/Packages/com.xrtk.core/package.json [hunk 1]
```diff
@@ -10,7 +10,7 @@
     "Mixed Reality",
     "DI"
   ],
-  "version": "0.2.11",
+  "version": "0.2.12",
```
Comment (StephenHodgson) [inline XRTK-Core/Packages/com.xrtk.core/package.json:13]: Let's revert the version until our next release

Comment (FejZa) [reply XRTK-Core/Packages/com.xrtk.core/package.json:13]: It's a breaking change and WMR won't be able to build without making a releas of core.

Comment (StephenHodgson) [reply XRTK-Core/Packages/com.xrtk.core/package.json:13]: Ah right, well that means we will need to do another release, but we just did one last night 😅

Comment (FejZa) [reply XRTK-Core/Packages/com.xrtk.core/package.json:13]: I am fine with holding of merging for a while and keeping the PR open. I am planning some more anyways.

**Comment by StephenHodgson** [comment]
Not sure what's up with this [Gradle daemon error](https://dev.azure.com/xrtk/Mixed%20Reality%20Toolkit/_build/results?buildId=6387&view=logs&j=0ab14b9f-e499-56d5-97b1-fd98b70ea339&t=859ff9fe-d969-518c-e56b-ea1c6c8313f3&l=807). maybe it has something to do with the new VMs?

**Comment by StephenHodgson** [comment]
Getting some missing profiles for the spatial awareness system
image

**Comment by StephenHodgson** [comment]
@FejZa could we also put the change to the spatial mesh collection into this PR as well since that is also a breaking change for Spatial Awareness?

**Review by StephenHodgson — APPROVED** [review]

**Pull request merged**